packages/zod/src/specific-issues.test.ts

```diff
@@ -0,0 +1,81 @@
+import { describe, expect, it } from "@jest/globals";
+import { expectType } from "@saiichihashimoto/test-utils";
+
+import {
+  defineArrayMember,
+  defineConfig,
+  defineField,
+  defineType,
+} from "@sanity-typed/types";
+import type { InferSchemaValues, PortableTextBlock } from "@sanity-typed/types";
+
+import { sanityConfigToZodsTyped } from "./internal";
+
+const fields: Omit<PortableTextBlock, "_type" | "children"> = {
+  level: 1,
+  listItem: "bullet",
+  markDefs: [],
+  style: "normal",
+};
+
+describe("specific issues", () => {
+  it("#559: zod parser strings untyped block markDefs", () => {
+    const config = defineConfig({
+      dataset: "dataset",
+      projectId: "projectId",
+      schema: {
+        types: [
+          defineType({
+            name: "foo",
+            type: "block",
+            marks: {
+              annotations: [
+                defineArrayMember({
+                  name: "internalLink",
+                  type: "object",
+                  fields: [
+                    defineField({
+                      name: "reference",
+                      type: "reference",
+                      to: [{ type: "post" as const }],
+                    }),
+                  ],
+                }),
+              ],
+            },
+          }),
+        ],
+      },
+    });
+    const zods = sanityConfigToZodsTyped(config);
+
+    const unparsed = {
+      ...fields,
+      _type: "foo",
+      children: [
+        { _key: "key", _type: "span", marks: ["mark-key"], text: "text" },
+      ],
+      markDefs: [
+        {
+          _key: "mark-key",
+          _type: "internalLink",
+          internalLink: {
+            _ref: "052a9ccc-ba1f-4e8e-adb4-4be196815746",
+            _type: "reference",
+          },
+          slug: {
+            _type: "slug",
+            current: "foo-bar-test",
+          },
+          postType: "post",
```

Comment:
Are these the keys that are getting stripped? Because that just means you need to add more fields to your `internalLink` object, not a `passthrough()`

Reply:
They aren't on the schema... these are resolved references from a groq query:
```
 {
  ...,
  // resolve referenced documents for proper routing
    body[]{
      ...,
      markDefs[]{
        ...,
        _type == "internalLink" => {
        "slug": @.internalLink->slug
      },
      "postType": @.internalLink->_type
    }
  },
}
```
I guess perhaps I'm asking zod to look the other way for a moment... I know I can't quite type all the annotations, but this will at least leave the fields in the response.

 Yeah, portable text is a beast to work with.